tests/e2e/logging/test_openshift-logging.py

```diff
@@ -171,3 +177,60 @@ def test_create_new_project_to_verify_logging(self, create_pvc_and_deploymentcon
         pod_obj.run_io(storage_type='fs', size=6000)
 
         self.validate_project_exists(pvc_obj)
+
+    @pytest.mark.polarion_id("OCS-650")
+    @tier4
+    def test_respin_osd_pods_to_verify_logging(self, create_pvc_and_deploymentconfig_pod):
+        """
+
+        """
+        # Create 1st project and app_pod
+        dc_pod_obj, dc_pvc_obj = create_pvc_and_deploymentconfig_pod
+
+        project1 = dc_pvc_obj.project.namespace
+
+        # Delete the OSD pod
+        disruption = disruption_helpers.Disruptions()
+        disruption.set_resource(resource='osd')
+        disruption.delete_resource()
+
+        # Check the health of the cluster-logging
+        assert ocp_logging_obj.check_health_of_clusterlogging()
+
+        # Check for the 1st project created in EFK stack after respin
+        pod_list = get_all_pods(namespace='openshift-logging')
+        elasticsearch_pod = [
+            pod.name for pod in pod_list if pod.name.startswith('elasticsearch')
+        ]
+        elasticsearch_pod_obj = get_pod_obj(
+            name=elasticsearch_pod[1], namespace='openshift-logging'
+        )
+        self.validate_project_exists(dc_pvc_obj)
+
+        # Check the files in the project
+        project1_filecount = elasticsearch_pod_obj.exec_cmd_on_pod(
+            command=f'es_util --query=project.{project1}.*/_count'
+        )
+        logger.info(f'File Count {project1_filecount}')
+        if project1_filecount:
```

Comment:
I see that this internally uses `exec_oc_cmd` function, which returns dict if true and string if false. So wouldn't this condition always be true?

Reply:
Fixed